clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java

```diff
@@ -226,6 +226,12 @@ public void promote(long blockId) throws IOException {
       // Nothing to promote
       return;
     }
+    for (BlockLocation loc : info.getLocations()) {
+      if (loc.getTierAlias().equals("mem")) {
```

Comment:
Is there a way to make this less hardcoded? (ie. what if the user only has ssd/hdd tiers) We can assume all workers have the same tier storage configurations.